backend/app/services/db/user_manager.py

```diff
@@ -0,0 +1,191 @@
+from passlib.context import CryptContext
+
+from .db_connect import connect
+
+
+def create_user(username: str, email: str, first_name: str, last_name: str,
+                password: str, bcrypt_context: CryptContext):
+    """Adds the specified user to the users database and adds the user's authentication details.
+
+    The user's entry in the authentication table is also created here to ensure atomicity.
+
+    Args:
+        username (str): The username associated with the user.
+        email (str): The email associated with the user.
+        first_name (str): The first name associated with the user.
+        last_name (str): The last name associated with the user.
+        password (str): The password that is used for authentication.
+        bcrypt_context (passlib.context.CryptContext): The hashing context
+        used for hashing the password.
+
+    Returns:
+        A dict with two keys, "status" and "message". Status is the status
+        of the user creation, either "success" or "failure".
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor()
+
+        hashed_password = bcrypt_context.hash(password)
+
+        user_query = '''
+        INSERT INTO user_information (username, email, first_name, last_name)
+        VALUES (%s, %s, %s, %s)
+        '''
+
+        auth_query = '''
+        INSERT INTO auth (username, password)
+        VALUES (%s, %s)
+        '''
+
+        user_values = (
+            username,
+            email,
+            first_name,
+            last_name
+        )
+
+        auth_values = (
+            username,
+            hashed_password
+        )
+
+        # Insert new user into users table
+        cursor.execute(user_query, user_values)
+        # Insert authentication details into authentication table
+        cursor.execute(auth_query, auth_values)
+        conn.commit()
+
+    return {
+        "message": f"New profile created for user {username}.",

```

Comment:
you literally say new profile created for user so in theory profile should be created at the same time as the user which means the user can't exist before creating the profile